Dart/src/com/jetbrains/lang/dart/ide/generation/DartGenerateToStringHandler.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2000-2016 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jetbrains.lang.dart.ide.generation;
+
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.util.Condition;
+import com.intellij.openapi.util.Pair;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.containers.ContainerUtil;
+import com.jetbrains.lang.dart.DartBundle;
+import com.jetbrains.lang.dart.DartComponentType;
+import com.jetbrains.lang.dart.psi.DartClass;
+import com.jetbrains.lang.dart.psi.DartComponent;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Map;
+
+public class DartGenerateToStringHandler extends BaseDartGenerateHandler {
+  @Override
+  protected String getTitle() {
+    return DartBundle.message("dart.generate.toString");
+  }
+
+  @Override
+  protected BaseCreateMethodsFix createFix(DartClass dartClass) {
+    return new CreateToStringFix(dartClass);
+  }
+
+  @Override
+  protected void collectCandidates(DartClass aClass, List<DartComponent> candidates) {
+    super.collectCandidates(aClass, candidates);
+  }
```

Comment:
> If we remove the contents of the first collectCandidates, the next collectCandidates is never called.

Yes, with current API of the `BaseDartGenerateHandler` class to optimize calculations we can completely override `collectCandidates(DartClass aClass, List<DartComponent> candidates)` which, for our case should only look for fields in the current class.
Or rewrite base class to be more convenient for all subclasses.